cli/tests/activate/hello.exp

```diff
@@ -6,62 +6,23 @@
 set dir [lindex $argv 0]
 set flox $env(FLOX_BIN)
 
-set timeout 5
-set cmd "$flox activate --dir $dir"
 spawn $flox activate --dir $dir
-expect {
-    "Building environment..." {}
-    timeout {
-      puts stderr "Reached timeout running '$cmd'"
-      exit 1
-    }
+expect_after {
+  timeout { exit 1 }
+  eof { exit 2 }
+  "*\n" { exp_continue }
 }
-expect {
-    -re "project.*\$" {}
-    timeout {
-      puts stderr "Reached timeout running '$cmd'"
-      exit 1
-    }
-}
-
-# activation is the only thing that should take very long so set a low timeout
-set timeout 1
+expect "Building"
+expect -re "flox .*\\\[project-\\d+\\\]" {}
```

Comment:
I'm a bit confused as to when we are using expect to "expect" certain strings and when we `assert_output` the _output_ in bats.
this one seems to do the former while `cli/tests/activate/hook.exp` is doing the latter

Reply:
Before sending commands we need to ensure the spawn is up and running. Possible the second expect here is not needed and can be removed.

Ideally we have minimal checking, especially for the finicky prompt stuff, in expect.